text_generation/causal_lm/cpp/continuous_batching/library/src/sampler.hpp

```diff
@@ -334,12 +336,12 @@ class Sampler {
         return logit_vector;
     }
 
-    int64_t _greedy_sample(const std::vector<LogitWithIdx>& logit_vector) const {
-        int64_t out_token = std::max_element(logit_vector.begin(), logit_vector.end(), [](const LogitWithIdx& lhs, const LogitWithIdx& rhs) { return lhs.first < rhs.first; }) - logit_vector.begin();
-        return out_token;
+    LogitWithIdx _greedy_sample(const std::vector<LogitWithIdx>& logit_vector) const {
+        auto out_token = std::max_element(logit_vector.begin(), logit_vector.end(), [](const LogitWithIdx& lhs, const LogitWithIdx& rhs) { return lhs.first < rhs.first; });
+        return *out_token;
     }
 
-    int64_t _multinomial_sample(const std::vector<LogitWithIdx>& logit_vector, float temperature, float top_p, size_t top_k) {
+    std::vector<LogitWithIdx> _multinomial_sample(const std::vector<LogitWithIdx>& logit_vector, float temperature, float top_p, size_t top_k, size_t num_tokens_per_sequence) {
```

Comment:
Does `num_tokens_per_sequence` refer to `SequenceGroup` or `Sequence`? 
If it's for the `SequenceGroup`, then can we rename it to `num_tokens_per_sequence_group` or `num_tokens_per_request` or something like that to be more explicit?

Reply:
maybe we can name it as `num_tokens_per_logit` ?

Reply:
Per Sequence. Please check the method to get logits per seq